cma/evolution_strategy.py

```diff
@@ -4646,4 +4661,35 @@ def update(es):
     es.objective_function_complements = [_al]
     es.augmented_lagrangian = _al
     es.best_feasible = best_feasible_solution
+
+    if post_optimization:
+        if h != no_constraints and (not isinstance(post_optimization, float) or post_optimization <= 0):
+            raise ValueError("In in case when equality constraints are present, "
```

Comment:
If we raise an error for this condition, we should do this in the beginning of the function. Think about a user who waited for, say, two hours for the optimization to finish and now gets this error instead of a potentially useful result of the optimization.